nomad/structs/structs.go

```diff
@@ -1775,6 +1776,51 @@ func (r *Resources) Merge(other *Resources) {
 	}
 }
 
+// COMPAT(0.10): Remove in 0.10
+func (r *Resources) Equals(o *Resources) bool {
+	if r == nil && o == nil {
+		return true
+	}
+	if r == nil || o == nil {
+		return false
+	}
```

Comment:
The two conditionals above are better expressed as:

```
if r == nil || o == nil {
  return r == o
}
```

The same pattern applies for nullness comparisons below.

Reply:
I've still got two conditionals, but now I should be able to short circuit the comparison if my pointers are == either because they're nil or point to the same object. Am I thinking about that right?